src/main/java/internet/store/web/filters/AuthorisationFilter.java

```diff
@@ -0,0 +1,73 @@
+package internet.store.web.filters;
+
+import internet.store.lib.Injector;
+import internet.store.model.Role;
+import internet.store.model.User;
+import internet.store.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorisationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("internet.store");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/order/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/create", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/all", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/product", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/user/order", Set.of(Role.RoleName.USER));
+    }
```

Comment:
What about these URLs: 
`/product/delete`
`/shopping-cart/product/add`
`/shopping-cart/product/remove`
`/order/create`
`/order/delete`
?

Reply:
Just Tested - you're right, from url I can reach to this menu even when access to the page with references to those methods is blocked...